### ISSUE TITLE
TOP PRIORITY: IDENTIFY STABLE ROLLBACK POINT

### ISSUE THREAD

**Issue by kylepolich** [data-skeptic/fast-spa#398]
The attached video demonstrates several critical issues that were introduced in the last few days.  We need a 100% FULL ROLLBACK asap.  I need to have the a key person able to login with out getting errors.

[bug.mp4.zip](https://github.com/data-skeptic/fast-spa/files/4365905/bug.mp4.zip)

The rollback needs to also resolve #396 and #390 by finding a time before these problems were introduced.

Save any changes you made and you can one-by-one submit new PRs once the code is fixed.

**Comment by Heartless49** [comment]
None of these are bugs that should involve a complete rollback; they were all related to the same underlying issue with navigation and fixed in https://github.com/data-skeptic/fast-spa/pull/399.

I recreated each of the situations in which there was a bug from this video, and each of them are resolved due to what the actual problem was.

- Back button taking you back to the dashboard.
![image](https://user-images.githubusercontent.com/1939701/77261903-2d830980-6c4f-11ea-8f9f-924810222cdb.png)
  - This was caused because the `createFolder` method was not adding a trailing `/` to the path, therefore the back button didn't realize you were in a directory, and took you back to the dashboard because it though you were trying to go to the directory outside of your instance's "Root" path.
- Creating a readme in a new folder and displaying the wrong readme file.
  - This was caused by the same issue from above in which the URL didn't have the trailing slash, causing forager to ignore the current directory as a "directory" and search for the readme of the parent directory. This has been fixed by the above resolution.
- README.md not populating the page after creation
  - This was caused because initially, the readme file is only retrieved when entering a directory - this has been changed so that when closing a file, and returning to the Prefix-page, it checks for a read me file.
